enterprise/models.py

```diff
@@ -1621,8 +1648,15 @@ def get_program(self, program_uuid):
         Return:
             dict: The program metadata.
         """
-        if not self.contains_programs([program_uuid]):
-            return None
+        request = crum.get_current_request()
+        # Temporarily gate enterprise catalog api usage behind waffle flag
+        if request and waffle.flag_is_active(request, USE_ENTERPRISE_CATALOG):
+            if not EnterpriseCatalogApiClient(user=request.user).contains_content_items(self.uuid, [program_uuid]):
+                return None
+        else:
+            if not self.contains_programs([program_uuid]):
+                return None
+
```

Comment:
Nit: the same basic chunk of code is repeated 4 times, maybe we can have a little helper method like
```python
def _check_content_exists(self, request, existence_checker, content_ids, null_value):
    # ....
```
...and you can invoke it like `return self._check_content_exists(request, self.contains_programs, [program_uuid], None)`, or `return self._check_content_exists(request, self.contains_courses, [course_run_id], (None, None))`.

Reply:
You're right it's definitely a bit repetitive. Given that we should be removing some of these code paths fairly soon with the rest of the catalog transition, I'm fine with leaving the duplication as it was before for the time being. But I also don't mind updating it if you'd rather

Reply:
I'm fine with leaving it as is, too, especially since this code is meant to be transient.